catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableFileSystemFileConsumer.java

```diff
@@ -38,36 +44,67 @@
   @Override
   protected boolean doPoll(String sha1) {
     if (observer != null) {
-      observer.addListener(listener);
+      observer.setListener(listener);
       observer.checkAndNotify();
-      observer.removeListener(listener);
-      fileSystemPersistenceProvider.store(sha1, observer);
+      observer.removeListener();
       return true;
     } else {
       return isMatched(null, null, null);
     }
   }
 
   @Override
-  protected void initialize(String fileName, String sha1) {
+  protected void initialize(String fileName) {
```

Comment:
❓What's the reason for not creating the `fileSystemPersistenceProvider` and `jsonSerializer` in the constructor instead of doing this here and checking for `null`?

Reply:
That's something I could not tell you even if I wanted. 🤷‍♂️ 
I can look into refactoring that